Bottom: 5036b5c3b236703f80b551bcef283a7d1f8343c5
Top:    ead4a19316ea90f45b643ab625170c80fb7951ac
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2017-02-19 09:09:37 +0000

Refresh of validate-and-swap-from-to

---

diff --git a/src/action.c b/src/action.c
index fb29d35..e3b0192 100644
--- a/src/action.c
+++ b/src/action.c
@@ -1277,6 +1277,7 @@ NotifyMode (void)
 	       Crosshair.AttachedLine.Point2.Y))
             {
               PinType *via;
+	      Cardinal last_layer_num, current_layer_num;
 	      Cardinal layer_from, layer_to;
 
               if ((line =
@@ -1300,6 +1301,9 @@ NotifyMode (void)
 		{
 		  layer_from = GetLayerNumber (PCB->Data, lastLayer);
 		  layer_to = GetLayerNumber (PCB->Data, CURRENT);
+
+                  layer_from = MIN (last_layer_num, current_layer_num);
+                  layer_to =   MAX (last_layer_num, current_layer_num);
 		}
 	      else
 		{
diff --git a/src/create.c b/src/create.c
index d41b456..c302a17 100644
--- a/src/create.c
+++ b/src/create.c
@@ -334,21 +334,8 @@ CreateNewViaEx (DataType *Data,
               DrillingHole, Name, Flags);
   if (Via)
     {
-      if (buried_from == buried_to)
-        {
-          Via->BuriedFrom = 0;
-          Via->BuriedTo = 0;
-	}
-      else if (buried_from <= buried_to)
-        {
-          Via->BuriedFrom = buried_from;
-          Via->BuriedTo = buried_to;
-	}
-      else
-        {
-          Via->BuriedFrom = buried_to;
-          Via->BuriedTo = buried_from;
-	}
+      Via->BuriedFrom = buried_from;
+      Via->BuriedTo = buried_to;
     }
   return Via;
 }
